modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAddressPermissions.testcase

```diff
@@ -0,0 +1,150 @@
+@component-name = "portal-commerce"
+definition {
+
+	property custom.properties = "jsonws.web.service.paths.excludes=${line.separator}company.security.strangers.verify=false";
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property release.feature.flags.disable.DISABLE_PRIVATE_LAYOUTS = "true";
+	property testray.main.component.name = "Order Management";
+
+	setUp {
+		CommerceConfiguration.commerceSetUp();
+
+        task ("Setup: Create an account with a buyer user") {
+			CommerceEntry.addAccountEntryUser(
+				accountName = "Commerce Account",
+				accountType = "Business",
+				agreeToTermsAndAnswerReminderQuery = "true",
+				createNewAccount = "true",
+				requireReset = "false",
+				userEmailAddress = "buyer@liferay.com",
+				userFirstName = "Buyer",
+				userLastName = "User",
+				userRole = "Buyer",
+				userScreenName = "buyeruser",
+				userSiteMembership = "Minium");
+		}
+
+		task ("Setup: 3 addresses are added to the account(Billing, Shipping, Billing & Shipping)") {
+			 Accounts.openAccountsAdmin();
+
+			LexiconEntry.gotoEntry(rowEntry = "Commerce Account");
+
+			Accounts.gotoAssigneeTab(assigneeTab = "Addresses");
+
+			for (var typeList : list "Billing,Shipping,Billing and Shipping") {
+				Accounts.addAddressViaSetDefaultAddress(
+					addressCity = "Diamond Bar",
+					addressCountry = "United States",
+					addressName = "Liferay ${typeList}",
+					addressPostalCode = "91765",
+					addressRegion = "California",
+					addressStreet1 = "1400 Montefino Ave",
+					addressType = "${typeList}",
+					phoneNumber = "+1-877-543-3729");
+     	   }
+		}
+
+		task ("Setup: The buyer doesn't have permission to manage addresses(Accounts > Account Entry: Manage Addresses)") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Users",
+				panel = "Control Panel",
+				portlet = "Roles");
+
+			Navigator.gotoNavItem(navItem = "Account Roles");
+
+			Search.searchCP(searchTerm = "Buyer");
+
+			Accounts.goToEntryMenuItem(
+				menuItem = "Edit",
+				rowEntry = "Buyer");
+
+			Navigator.gotoNavItem(navItem = "Define Permissions");
+
+			Type(
+				locator1 = "RolesPermissionsNavigation#SEARCH_FIELD",
+				value1 = "Accounts");
+
+			ClickNoError(
+				locator1 = "RolesPermissionsNavigation#PERMISSIONS_NAVITEM_SUBITEM",
+				key_navItem = "Accounts",
+				key_subItem = "Accounts");
+				
+			Uncheck.uncheckNotVisible(
+				locator1 = "RolesPermissionsNavigation#PERMISSIONS_ITEM_CHECKBOX",
+				key_itemName = "Account Entry",
+				key_checkboxName = "MANAGE_ADDRESSES");
+
+			Button.clickSave();
+		}

```

Comment:
You can replace all these steps with one json macro:

Permissions.removePermissionViaAPI(
				groupName = "Minium",
				resourceAction = "MANAGE_ADDRESSES",
				resourceName = "com.liferay.account.model.AccountEntry",
				roleTitle = "Buyer",
				roleType = "account");